src/SwiftUI/SwiftUI/Color.cs

```diff
@@ -0,0 +1,226 @@
+
+		static Color? red = null;
+		public static Color Red => red ??= new Color (GetColorRed (0));
+
+		public static Color Secondary => new Color (GetColorSecondary (0));
+
+		static Color? white = null;
+		public static Color White => white ??= new Color (GetColorWhite (0));
+
+		static Color? yellow = null;
+		public static Color Yellow => yellow ??= new Color (GetColorYellow (0));
+		#endregion
+
+		protected override void InitNativeData (void* handle)
+		{
+			// Using TransferFuncType.InitWithTake here in case we leak.
+			using (var dataHandle = Data.GetSwiftHandle ())
+				SwiftType.Transfer (handle, dataHandle.Pointer, TransferFuncType.InitWithTake);
+		}
+
+		#region Constructors
+		internal Color (IntPtr data)
+		{
+			Data = data;
+		}
+
+		public Color (RGBColorSpace colorSpace, double red, double green, double blue, double opacity)
+		{
+			Data = CreateFromRGBColorSpaceRedGreenBlueOpacity ((ulong)colorSpace, red, green, blue, opacity);
+		}
+
+		public Color (RGBColorSpace colorSpace, double white, double opacity)
+		{
+			Data = CreateFromRGBColorSpaceWhiteOpacity ((ulong)colorSpace, white, opacity);
+        }
+
+        public Color (double hue, double saturation, double brightness, double opacity)
+		{
+			Data = CreateFromHSBO (hue, saturation, brightness, opacity);
+		}
+		#endregion
+
+		#region DllImports
+
+		#region Initialisers
+		[DllImport (SwiftUILib.Path,
+			CallingConvention = CallingConvention.Cdecl,
+			EntryPoint = "$s7SwiftUI5ColorV_3red5green4blue7opacityA2C13RGBColorSpaceO_S4dtcfC")]
+		static extern IntPtr CreateFromRGBColorSpaceRedGreenBlueOpacity (
+			ulong colourSpace,
+			double red,
+			double green,
+			double blue,
+			double opacity);
+
+		[DllImport (SwiftUILib.Path,
+			CallingConvention = CallingConvention.Cdecl,
+			EntryPoint = "$s7SwiftUI5ColorV_5white7opacityA2C13RGBColorSpaceO_S2dtcfC")]
+		static extern IntPtr CreateFromRGBColorSpaceWhiteOpacity (
+			ulong colourSpace,
+			double white,
+			double opacity);
+
+
+		[DllImport (SwiftUILib.Path,
+			CallingConvention = CallingConvention.Cdecl,
+			EntryPoint = "$s7SwiftUI5ColorV3hue10saturation10brightness7opacityACSd_S3dtcfC")]
+		static extern IntPtr CreateFromHSBO (
+			double hue,
+			double saturation,
+			double brightness,
+			double opacity);
+		#endregion
+
+		#region RGBColorSpace
+		[DllImport (SwiftUILib.Path,
+			CallingConvention = CallingConvention.Cdecl,
+			EntryPoint = "$s7SwiftUI5ColorV13RGBColorSpaceO9displayP3yA2EmFWC")]
+		static extern IntPtr GetColorSpaceDisplayP3 (TypeMetadata* colorSpaceMetadata);
+
+		[DllImport (SwiftUILib.Path,
+			CallingConvention = CallingConvention.Cdecl,
+			EntryPoint = "$s7SwiftUI5ColorV13RGBColorSpaceO4sRGByA2EmFWC")]
+		static extern IntPtr GetColorSpaceRGB ();
+
+		[DllImport (SwiftUILib.Path,
+			CallingConvention = CallingConvention.Cdecl,
+			EntryPoint = "$s7SwiftUI5ColorV13RGBColorSpaceO10sRGBLinearyA2EmFWC")]
+		static extern IntPtr GetColorSpaceRGBLinear ();
```

Comment:
It looks like this is missing the metadata argument?

Reply:
As it looks like these ColorSpaces will take a little time to workout, I've removed all the `RGBColorSpace` constructors for now, so a working version can go in. Other constructors have been tested and work.